### ISSUE TITLE
quickstart_finetune.py requires the OpenAI Api key even though it doesn't use the gpt model.

### ISSUE THREAD

**Issue by berkay3500** [anarchy-ai/LLM-VM#289]
I use "pythia" as the large model and "opt" as the small model, I deleted openai_key arg but it throws this error;
```
Downloading (…)lve/main/config.json: 100%|███████████████████████████████████████████████| 644/644 [00:00<00:00, 1.88MB/s]
Downloading pytorch_model.bin: 100%|███████████████████████████████████████████████████| 663M/663M [04:30<00:00, 2.45MB/s]
Downloading (…)neration_config.json: 100%|████████████████████████████████████████████████| 137/137 [00:00<00:00, 344kB/s]
Downloading (…)okenizer_config.json: 100%|███████████████████████████████████████████████| 685/685 [00:00<00:00, 1.33MB/s]
Downloading (…)olve/main/vocab.json: 100%|█████████████████████████████████████████████| 899k/899k [00:00<00:00, 1.54MB/s]
Downloading (…)olve/main/merges.txt: 100%|█████████████████████████████████████████████| 456k/456k [00:00<00:00, 2.02MB/s]
Downloading (…)cial_tokens_map.json: 100%|███████████████████████████████████████████████| 441/441 [00:00<00:00, 1.19MB/s]
Using model: pythia
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:0 for open-end generation.
Exception in thread Thread-3 (new_thread):
{'completion': 'Q.\n\nHow to use the currency in myanmmarAnswer question Q.Q.\n\nHow to use the currency in myanmmarAnswer question Q.Q.\n\nHow to use the currency in myanmmarAnswer question Q.Q.\n\nHow to use the currency in myanmmarAnswer question Q.Q.\n\nHow to use the currency in myanmmarAnswer question Q.Q.\n\nHow to use the currency in myanmmarAnswer question Q.Q.\n\nHow to use the currency in myanmmarAnswer question Q.Q.\n\nHow to use the currency in myanmmarAnswer question Q.Q.\n\nHow to use the currency in myanmmarAnswer question Q.Q.\n\nHow to use the currency in myanmmarAnswer question Q.Q.\n\nHow to use the currency in myanmmarAnswer question Q.Q.\n\nHow to use the currency in myanmmarAnswer question Q.Q.\n\nHow to use the currency in myanmmarAnswer question Q.Q.\n\nHow to use', 'status': 200}
Traceback (most recent call last):
  File "/home/berkay/anaconda3/envs/anarchy/lib/python3.10/threading.py", line 1009, in _bootstrap_inner
    self.run()
  File "/home/berkay/anaconda3/envs/anarchy/lib/python3.10/threading.py", line 946, in run
    self._target(*self._args, **self._kwargs)
  File "/home/berkay/anaconda3/envs/anarchy/lib/python3.10/site-packages/llm_vm/completion/optimize.py", line 44, in new_thread
    t[0] = foo()
  File "/home/berkay/anaconda3/envs/anarchy/lib/python3.10/site-packages/llm_vm/completion/optimize.py", line 301, in promise
    return succeed_train(use_completion)
  File "/home/berkay/anaconda3/envs/anarchy/lib/python3.10/site-packages/llm_vm/completion/optimize.py", line 275, in actual_train
    for j in self.data_synthesizer.data_synthesis(self,prompt,best_completion,openai_key=self.openai_key, regex = regex, type = type, choices = choices, **kwargs):
  File "/home/berkay/anaconda3/envs/anarchy/lib/python3.10/site-packages/llm_vm/completion/data_synthesis.py", line 42, in data_synthesis
    datapoint = self.generate_example(final_prompt, openai_key, regex = regex, type = type, choices = choices)
  File "/home/berkay/anaconda3/envs/anarchy/lib/python3.10/site-packages/llm_vm/completion/data_synthesis.py", line 97, in generate_example
    response = openai.ChatCompletion.create(messages=cur_prompt,model=model,max_tokens=max_tokens,temperature=temperature)['choices'][0]['message']['content']
  File "/home/berkay/anaconda3/envs/anarchy/lib/python3.10/site-packages/openai/api_resources/chat_completion.py", line 25, in create
    return super().create(*args, **kwargs)
  File "/home/berkay/anaconda3/envs/anarchy/lib/python3.10/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 149, in create
    ) = cls.__prepare_create_request(
  File "/home/berkay/anaconda3/envs/anarchy/lib/python3.10/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 106, in __prepare_create_request
    requestor = api_requestor.APIRequestor(
  File "/home/berkay/anaconda3/envs/anarchy/lib/python3.10/site-packages/openai/api_requestor.py", line 138, in __init__
    self.api_key = key or util.default_api_key()
  File "/home/berkay/anaconda3/envs/anarchy/lib/python3.10/site-packages/openai/util.py", line 186, in default_api_key
    raise openai.error.AuthenticationError(
openai.error.AuthenticationError: No API key provided. You can set your API key in code using 'openai.api_key = <API-KEY>', or you can set the environment variable OPENAI_API_KEY=<API-KEY>). If your API key is stored in a file, you can point the openai module at it with 'openai.api_key_path = <PATH>'. You can generate API keys in the OpenAI web interface. See https://platform.openai.com/account/api-keys for details.
```

**Comment by daspartho** [comment]
I believe this is caused due to the `generate_example` function here which calls the openai API regardless of the given `big_model` as it does not have support for other models.

https://github.com/anarchy-ai/LLM-VM/blob/05d771b467970cfd0fa9bd191d409cf2c00ba37a/src/llm_vm/completion/data_synthesis.py#L50-L54

**Comment by daspartho** [comment]
If we replace line 54 with `optimizer.big_model.generate()` it'll work. @cartazio @VictorOdede can you confirm?

I'll make a PR to fix it!

**Comment by cartazio** [comment]
that looks correct :(

**Comment by cartazio** [comment]
(in that we clearly have a bug there)

**Comment by daspartho** [comment]
alright- will work on a PR for this! could you assign it to me?

**Comment by daspartho** [comment]
opened a PR to fix it #299

**Comment by daspartho** [comment]
**update**: generating examples for finetuning can't be done using other big models yet since only chatGPT can reliably mimic prompts (for now)- so the previous PR to support all big models was reverted.

https://github.com/anarchy-ai/LLM-VM/pull/299#issuecomment-1769094312
